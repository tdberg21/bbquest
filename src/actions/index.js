export const addRestaurants = (restaurants) => ({
  type: 'ADD_RESTAURANTS',
  restaurants
});

export const loginUser = (username, id) => ({
  type: 'LOGIN_USER',
  username,
  id
});

