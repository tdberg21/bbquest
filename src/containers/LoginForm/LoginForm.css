.login-form-container {
  background-image: url(https://www.pigrigbbq.com/files/2017-02/background.jpg);
  height: 100vh;
  padding-top: 1rem;
  width: 101%;
  background-size: cover;
}

.login-form {
  background-color: rgb(243, 243, 243, .8);
  margin: 5em auto;
  width: 40%;
  height: 18rem;
  padding: 1.5rem;
  border: 1px solid #333;
  border-radius: 1em;
}

label {
  text-align: left;
  font-weight: 600;
}

.inputs {
  display: flex;
  flex-direction: column;
  width: 75%;
  margin: auto; 
}

.input-fields {
  color: #333;
  display: block;
  font-size: 1em;
  font-weight: 300;
  margin: 0.5rem auto;
  padding: .5rem;
  width: 80%;
  height: 0.4em;
}

.form-header {
  margin: 0.5rem auto;
  font-size: 1rem;
}

.login-button {
  background-color: #333;
  color: #fff;
  width: 86%;
  margin: 1rem auto;
  height: 2.5em;
}