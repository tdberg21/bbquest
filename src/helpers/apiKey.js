/*eslint-disable max-len*/

export const apiKey = '13BQs-tn_WMJ5SaCTsASZO80Yd4WuC8TrjH-CTXyx_k236k_1VvdI4NfTev4AovYBj7xoA1KnCBbrbiWXHYhB2saFILNKrL2kyfRQwAFZN8ks_LOODJwZ4dMqn5PW3Yx';