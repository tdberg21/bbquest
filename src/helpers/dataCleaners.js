export const scrubRestaurants =(response) => (
  response.businesses.map((business) => {
    return {
      name: business.name,
      rating: business.rating,
      price: business.price
    };
  }));