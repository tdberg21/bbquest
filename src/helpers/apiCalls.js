import { apiKey } from './apiKey';

export const fetchRestaurantData = async (key, location) => {
  const corsAnywhereUrl = 'https://cors-anywhere.herokuapp.com/';
  const url = `${corsAnywhereUrl}https://api.yelp.com/v3/businesses/search?location=${location}&categories=bbq&limit=15`;
  const response = await fetch(url, {
    method: 'GET',
    headers: {
      'Authorization': `Bearer ${key}`,
      'Content-Type': 'application/json'
    }
  });
  const results = await response.json();
  return results;
};

export const fetchDetails = async (id) => {
  const corsAnywhereUrl = 'https://cors-anywhere.herokuapp.com/';
  const url = `${corsAnywhereUrl}https://api.yelp.com/v3/businesses/${id}/reviews`;
  const response = await fetch(url, {
    method: 'GET',
    headers: {
      'Authorization': `Bearer ${apiKey}`,
      'Content-Type': 'application/json'
    }
  });
  const results = await response.json();
  return results;
}; 

export const signUpUser = async (username, email, password) => {
  const prefix = `http://localhost:3000/api/v1/users/new`;
  const suffix = `?username=${username}&email=${email}&password=${password}`;
  const url = `${prefix}${suffix}`;
  const response = await fetch(url, {
    method: 'POST',
    body: JSON.stringify({
      username,
      email,
      password
    }),
    headers: {
      'Content-Type': 'application/json'
    }
  });
  const results = await response.json();
  return await results; 
};

export const fetchUser = async (username, email, password) => {
  const url = `http://localhost:3000/api/v1/users`;
  const response = await fetch(url, {
    method: 'POST',
    body: JSON.stringify({
      username,
      email,
      password
    }),
    headers: {
      'Content-Type': 'application/json'
    }
  });
  const results = await response.json();
  return await results; 
};

export const fetchVisitedRestaurants = async (user_id) => {
  const url = `http://localhost:3000/api/v1/restaurants/?id=${user_id}`;
  const response = await fetch(url);
  const results = await response.json();
  return await results;
};

export const addVisitedRestaurant = async (rating, notes, date, user_id, restaurant_name, meal, yelpId) => {
  // if (!rating || rating === '') {
  //   rating = 'n/a';
  // }
  // if (!notes) {
  //   notes='n/a';
  // }
  // if (!date) {
  //   date='n/a';
  // }

  const url = `http://localhost:3000/api/v1/restaurants?rating=${rating}&notes=${notes}&date=${date}&user_id=${user_id}&restaurant_name=${restaurant_name}&meal=${meal}&yelpId=${yelpId}`;
  const response = await fetch(url, {
    method: 'POST',
    body: JSON.stringify({
      rating,
      notes,
      date,
      user_id,
      restaurant_name,
      meal,
      yelpId
    })});
  const results = await response.json();
  return await results;
};