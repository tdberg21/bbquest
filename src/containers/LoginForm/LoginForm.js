import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { loginUser, addVisited } from '../../actions';
import { fetchUser, fetchVisitedRestaurants } from '../../helpers/apiCalls';
import './LoginForm.css';
import SearchForm from '../SearchForm/SearchForm';
import PropTypes from 'prop-types';

export class LoginForm extends Component {
  constructor() {
    super();

    this.state = {
      username: '',
      email: '',
      password: ''
    };
  }

  handleChange = (event) => {
    const { name, value } = event.target;
    this.setState({
      [name]: value
    });
  }

  handleSubmit = async (event) => {
    event.preventDefault();
    const { username, email, password } = this.state;
    const results = await fetchUser(username, email, password);
    const visited = await fetchVisitedRestaurants(results.id);
    this.props.loginUser(results.username, results.id);
    this.props.addVisited(visited);
    this.props.history.push('/search');
  }

  resetState = () => {
    this.setState({
      username: '',
      email: '',
      password: ''
    });
  }

  render() {
    if (!this.props.username) {
      return (
        <div className='login-form-container'>
          <form 
            className='login-form login-form-test'
            onSubmit={this.handleSubmit}
          >
            <div className='inputs'>
              <label htmlFor="username">Username:</label>
              <input
                id='username'
                className='username-field input-fields'
                aria-label='Please Enter Your Username'
                type='text'
                name='username'
                value={this.state.username}
                onChange={this.handleChange}
              />
              <label htmlFor="email">Email:</label>
              <input
                id='email'
                className='email-field input-fields'
                aria-label='Please Enter Your Email'
                type='text'
                name='email'
                value={this.state.email}
                onChange={this.handleChange}
              />
              <label htmlFor="password">Password:</label>
              <input
                className='password-field input-fields'
                aria-label='Please Enter Your Password'
                type='password'
                name='password'
                value={this.state.password}
                onChange={this.handleChange}
              />
              <button
                className='login-button'
                aria-label='Sign in to your account'
              >
                Login
              </button>
            </div>
            <Link to='/signup'>
              Don't have an account?
            </Link>
          </form>
        </div>
      );
    } else {
      return (
        <div>
          <SearchForm />
        </div>
      );
    }
  }
}

export const mapStateToProps = (state) => ({
  username: state.user.username
});

export const mapDispatchToProps = (dispatch) => {
  return {
    loginUser: (username, id) => dispatch(loginUser(username, id)),
    addVisited: (visited) => dispatch(addVisited(visited))
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(LoginForm);

LoginForm.propTypes = {
  username: PropTypes.string,
  addVisited: PropTypes.func,
  loginUser: PropTypes.func,
  history: PropTypes.object
};