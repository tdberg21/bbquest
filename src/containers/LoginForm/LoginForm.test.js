import React from 'react';
import { shallow } from 'enzyme';
import { LoginForm, mapDispatchToProps } from './LoginForm';
import { loginUser, addVisited } from '../../actions';
import { fetchUser } from '../../helpers/apiCalls';

describe('LOGINFORM TESTS', () => {
  let wrapper;
  let mockHistory = {push: jest.fn()};

  beforeEach(() => {
    wrapper = shallow(<LoginForm loginUser={jest.fn()} addVisited={jest.fn()} history={mockHistory}/>);
  });

  it('should match the snapshot without a user', () => {  
    wrapper = shallow(<LoginForm username='Tory'/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot with a user', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update the state when handleChange is invoked', () => {
    const mockEvent = { target: { value: 'abc', name: 'username' } };

    wrapper.instance().handleChange(mockEvent);
    const results = wrapper.state('username');

    expect(results).toEqual('abc');
  });


  it('should invoke handleSubmit when the form is submitted', () => {
    const spy = jest.spyOn(wrapper.instance(), 'handleSubmit');
    wrapper.setState({
      username: 'nick',
      email: 'nick@msn.com',
      password: 'unleashthecage'
    });
    const mockEvent = { preventDefault: jest.fn() };
    const submitButton = wrapper.find('.login-form-test');
    
    submitButton.simulate('submit', mockEvent);

    expect(spy).toHaveBeenCalledWith(mockEvent);
  });

  it('should invoke fetchUser when handleSubmit is called', async () => {
    const mockEvent = { preventDefault: jest.fn() };
    await wrapper.instance().handleSubmit(mockEvent);

    expect(fetchUser).toHaveBeenCalled();
  });

  it('should invoke fetchVisitedRestaurants when handleSubmit is called', async () => {
    const mockEvent = { preventDefault: jest.fn() };
    await wrapper.instance().handleSubmit(mockEvent);

    expect(fetchVisitedRestaurants).toHaveBeenCalled();
  });

  it('should reset the state when resetState is invoked', () => {
    const expected = {username: '', email: '', password: ''};
    wrapper.instance().setState({
      username: 'Todd',
      email: 'todd@a.com',
      password: 'password'
    });
    wrapper.instance().resetState();
    const results = wrapper.state();
  
    expect(results).toEqual(expected);
  });


  describe('mapDispatchToProps', () => {
    it('calls dispatch with loginUser action', async () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = loginUser('Tory', 9);
      const mappedProps = mapDispatchToProps(mockDispatch);

      mappedProps.loginUser('Tory', 9);

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });

    it('calls dispatch with addVisited action', async () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = addVisited('PeeWees');
      const mappedProps = mapDispatchToProps(mockDispatch);

      mappedProps.addVisited('PeeWees');

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
});