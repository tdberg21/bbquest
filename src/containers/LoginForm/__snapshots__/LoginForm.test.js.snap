// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LOGINFORM TESTS should match the snapshot with a user 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginForm
    addVisited={[Function]}
    history={
        Object {
            "push": [Function],
          }
    }
    loginUser={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        className="login-form login-form-test"
        onSubmit={[Function]}
>
        <div
                className="inputs"
        >
                <label
                        htmlFor="username"
                >
                        Username:
                </label>
                <input
                        aria-label="Please Enter Your Username"
                        className="username-field input-fields"
                        id="username"
                        name="username"
                        onChange={[Function]}
                        type="text"
                        value=""
                />
                <label
                        htmlFor="email"
                >
                        Email:
                </label>
                <input
                        aria-label="Please Enter Your Email"
                        className="email-field input-fields"
                        id="email"
                        name="email"
                        onChange={[Function]}
                        type="text"
                        value=""
                />
                <label
                        htmlFor="password"
                >
                        Password:
                </label>
                <input
                        aria-label="Please Enter Your Password"
                        className="password-field input-fields"
                        name="password"
                        onChange={[Function]}
                        type="password"
                        value=""
                />
                <button
                        aria-label="Sign in to your account"
                        className="login-button"
                >
                        Login
                </button>
        </div>
        <Link
                replace={false}
                to="/signup"
        >
                Don't have an account?
        </Link>
</form>,
      "className": "login-form-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="inputs"
>
            <label
                        htmlFor="username"
            >
                        Username:
            </label>
            <input
                        aria-label="Please Enter Your Username"
                        className="username-field input-fields"
                        id="username"
                        name="username"
                        onChange={[Function]}
                        type="text"
                        value=""
            />
            <label
                        htmlFor="email"
            >
                        Email:
            </label>
            <input
                        aria-label="Please Enter Your Email"
                        className="email-field input-fields"
                        id="email"
                        name="email"
                        onChange={[Function]}
                        type="text"
                        value=""
            />
            <label
                        htmlFor="password"
            >
                        Password:
            </label>
            <input
                        aria-label="Please Enter Your Password"
                        className="password-field input-fields"
                        name="password"
                        onChange={[Function]}
                        type="password"
                        value=""
            />
            <button
                        aria-label="Sign in to your account"
                        className="login-button"
            >
                        Login
            </button>
</div>,
          <Link
            replace={false}
            to="/signup"
>
            Don't have an account?
</Link>,
        ],
        "className": "login-form login-form-test",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                htmlFor="username"
>
                Username:
</label>,
              <input
                aria-label="Please Enter Your Username"
                className="username-field input-fields"
                id="username"
                name="username"
                onChange={[Function]}
                type="text"
                value=""
/>,
              <label
                htmlFor="email"
>
                Email:
</label>,
              <input
                aria-label="Please Enter Your Email"
                className="email-field input-fields"
                id="email"
                name="email"
                onChange={[Function]}
                type="text"
                value=""
/>,
              <label
                htmlFor="password"
>
                Password:
</label>,
              <input
                aria-label="Please Enter Your Password"
                className="password-field input-fields"
                name="password"
                onChange={[Function]}
                type="password"
                value=""
/>,
              <button
                aria-label="Sign in to your account"
                className="login-button"
>
                Login
</button>,
            ],
            "className": "inputs",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Username:",
                "htmlFor": "username",
              },
              "ref": null,
              "rendered": "Username:",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "aria-label": "Please Enter Your Username",
                "className": "username-field input-fields",
                "id": "username",
                "name": "username",
                "onChange": [Function],
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Email:",
                "htmlFor": "email",
              },
              "ref": null,
              "rendered": "Email:",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "aria-label": "Please Enter Your Email",
                "className": "email-field input-fields",
                "id": "email",
                "name": "email",
                "onChange": [Function],
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Password:",
                "htmlFor": "password",
              },
              "ref": null,
              "rendered": "Password:",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "aria-label": "Please Enter Your Password",
                "className": "password-field input-fields",
                "name": "password",
                "onChange": [Function],
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "aria-label": "Sign in to your account",
                "children": "Login",
                "className": "login-button",
              },
              "ref": null,
              "rendered": "Login",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Don't have an account?",
            "replace": false,
            "to": "/signup",
          },
          "ref": null,
          "rendered": "Don't have an account?",
          "type": [Function],
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          className="login-form login-form-test"
          onSubmit={[Function]}
>
          <div
                    className="inputs"
          >
                    <label
                              htmlFor="username"
                    >
                              Username:
                    </label>
                    <input
                              aria-label="Please Enter Your Username"
                              className="username-field input-fields"
                              id="username"
                              name="username"
                              onChange={[Function]}
                              type="text"
                              value=""
                    />
                    <label
                              htmlFor="email"
                    >
                              Email:
                    </label>
                    <input
                              aria-label="Please Enter Your Email"
                              className="email-field input-fields"
                              id="email"
                              name="email"
                              onChange={[Function]}
                              type="text"
                              value=""
                    />
                    <label
                              htmlFor="password"
                    >
                              Password:
                    </label>
                    <input
                              aria-label="Please Enter Your Password"
                              className="password-field input-fields"
                              name="password"
                              onChange={[Function]}
                              type="password"
                              value=""
                    />
                    <button
                              aria-label="Sign in to your account"
                              className="login-button"
                    >
                              Login
                    </button>
          </div>
          <Link
                    replace={false}
                    to="/signup"
          >
                    Don't have an account?
          </Link>
</form>,
        "className": "login-form-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="inputs"
>
              <label
                            htmlFor="username"
              >
                            Username:
              </label>
              <input
                            aria-label="Please Enter Your Username"
                            className="username-field input-fields"
                            id="username"
                            name="username"
                            onChange={[Function]}
                            type="text"
                            value=""
              />
              <label
                            htmlFor="email"
              >
                            Email:
              </label>
              <input
                            aria-label="Please Enter Your Email"
                            className="email-field input-fields"
                            id="email"
                            name="email"
                            onChange={[Function]}
                            type="text"
                            value=""
              />
              <label
                            htmlFor="password"
              >
                            Password:
              </label>
              <input
                            aria-label="Please Enter Your Password"
                            className="password-field input-fields"
                            name="password"
                            onChange={[Function]}
                            type="password"
                            value=""
              />
              <button
                            aria-label="Sign in to your account"
                            className="login-button"
              >
                            Login
              </button>
</div>,
            <Link
              replace={false}
              to="/signup"
>
              Don't have an account?
</Link>,
          ],
          "className": "login-form login-form-test",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <label
                  htmlFor="username"
>
                  Username:
</label>,
                <input
                  aria-label="Please Enter Your Username"
                  className="username-field input-fields"
                  id="username"
                  name="username"
                  onChange={[Function]}
                  type="text"
                  value=""
/>,
                <label
                  htmlFor="email"
>
                  Email:
</label>,
                <input
                  aria-label="Please Enter Your Email"
                  className="email-field input-fields"
                  id="email"
                  name="email"
                  onChange={[Function]}
                  type="text"
                  value=""
/>,
                <label
                  htmlFor="password"
>
                  Password:
</label>,
                <input
                  aria-label="Please Enter Your Password"
                  className="password-field input-fields"
                  name="password"
                  onChange={[Function]}
                  type="password"
                  value=""
/>,
                <button
                  aria-label="Sign in to your account"
                  className="login-button"
>
                  Login
</button>,
              ],
              "className": "inputs",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Username:",
                  "htmlFor": "username",
                },
                "ref": null,
                "rendered": "Username:",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "aria-label": "Please Enter Your Username",
                  "className": "username-field input-fields",
                  "id": "username",
                  "name": "username",
                  "onChange": [Function],
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Email:",
                  "htmlFor": "email",
                },
                "ref": null,
                "rendered": "Email:",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "aria-label": "Please Enter Your Email",
                  "className": "email-field input-fields",
                  "id": "email",
                  "name": "email",
                  "onChange": [Function],
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Password:",
                  "htmlFor": "password",
                },
                "ref": null,
                "rendered": "Password:",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "aria-label": "Please Enter Your Password",
                  "className": "password-field input-fields",
                  "name": "password",
                  "onChange": [Function],
                  "type": "password",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "aria-label": "Sign in to your account",
                  "children": "Login",
                  "className": "login-button",
                },
                "ref": null,
                "rendered": "Login",
                "type": "button",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Don't have an account?",
              "replace": false,
              "to": "/signup",
            },
            "ref": null,
            "rendered": "Don't have an account?",
            "type": [Function],
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`LOGINFORM TESTS should match the snapshot without a user 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginForm
    username="Tory"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Connect(SearchForm) />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Connect(SearchForm) />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
